Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -735,6 +735,31 @@ public extension _ChatChannelController {
         }
     }
 
+    /// Pin the message with the provided message id.
+    ///  - Parameters:
+    ///   - messageId: The message identifier.
+    ///   - pin: The pinning expiration information. It supports setting an infinite expiration, setting a date, or the amount of time a message is pinned.
+    ///   - completion: A completion block with an error if the request was failed.
+    func pinMessage(messageId: MessageId, _ pin: Pin, completion: ((Error?) -> Void)? = nil) {
+        updater.pinMessage(messageId: messageId, pin) { result in
+            self.callback {
+                completion?(result)
+            }
+        }
+    }
+
+    /// Unpin the message with the provided message id.
+    ///  - Parameters:
+    ///   - messageId: The message identifier.
+    ///   - completion: A completion block with an error if the request was failed.
+    func unpinMessage(messageId: MessageId, completion: ((Error?) -> Void)? = nil) {
+        updater.unpinMessage(messageId: messageId) { result in
+            self.callback {
+                completion?(result)
+            }
+        }
+    }
+
```

Comment:
This modifies the messages so shouldn't this rather be on `MessageController`?

Reply:
In the documentation all other platforms pin and unpin the message in the channelController. So this makes it consistent with the other platforms, and also kinda makes sense, because we pinning a message in a channel, and not pinning a message in all channels, don't know if this last sentence made sense 😄